lib/moon_web/pages/components/input_group_page.ex

```diff
@@ -0,0 +1,328 @@
+                  for={@user_changeset}
+                  change="register_form_update"
+                  submit="register_form_submit"
+                  autocomplete="off"
+                >
+                  <InputGroup :let={input_group_class: input_group_class} has_fields>
+                    <Field name={:email} class={input_group_class}>
+                      <TextInput size="xl" type="email" placeholder="Email" />
+                    </Field>
+                    <Field name={:password} class={input_group_class}>
+                      <TextInput size="xl" type="password" id="password3" placeholder="Password" />
+                    </Field>
+                  </InputGroup>
+
+                  <div class="pt-4 flex items-center justify-around">
+                    <Button type="submit" right_icon="arrows_right" variant="primary">Register</Button>
+                  </div>
+                </Form>
+              </TopToDown>
+            </:example>
+
+            <:code>{input_group_1000_code()}</:code>
+
+            <:state>{input_group_1000_state(assigns)}</:state>
+          </ExampleAndCode>
+        </Context>
+
+        <div>
+          <div class="text-bulma-100 items-center text-moon-20 font-normal my-4">Input Group Props</div>
+          <Table items={@props_info_array}>
+            <Column name="name" label="Name" :let={item: item} is_row_header>
+              {item.name}
+            </Column>
+            <Column name="type" label="Type" :let={item: item}>
+              {item.type}
+            </Column>
+            <Column name="required" label="Required" :let={item: item}>
+              {item.required}
+            </Column>
+            <Column name="default" label="Default" :let={item: item}>
+              {item.default}
+            </Column>
+            <Column name="description" label="Description" :let={item: item}>
+              {item.description}
+            </Column>
+          </Table>
+        </div>
+      </TopToDown>
+    </Page>
+    """
+  end
+
+  defp input_group_1_code() do
+    """
+    <LeftToRight class="items-center justify-around w-full items-end">
+      <InputGroup :let={input_group_class: input_group_class}>
+        <TextInput size="xl" placeholder="Country" class={input_group_class} />
+        <TextInput size="xl" placeholder="Phone" class={input_group_class} />
+      </InputGroup>
+
+      <InputGroup orientation="vertical" :let={input_group_class: input_group_class}>
+        <TextInput size="xl" placeholder="Country" class={input_group_class} />
+        <TextInput size="xl" placeholder="Phone" class={input_group_class} />
+      </InputGroup>
+    </LeftToRight>
+    """
+  end
+
+  defp input_group_10_code() do
+    """
+    <LeftToRight class="items-center justify-around w-full items-end">
+      <InputGroup dir="rtl" :let={input_group_class: input_group_class}>
+        <TextInput size="xl" placeholder="Country" class={input_group_class} dir="rtl" />
```

Comment:
This passing input_group_class looks complex and let's try to find a way to simplify this.

Reply:
OK. I have been looking for a way since yesterday. I will try to find more.